### ISSUE TITLE
Abalone Age Prediction using various ML classifiers

### ISSUE THREAD

**Issue by vrn25** [WebClub-NITK/Hacktoberfest-2k19#27]
### Description
Given **Multi-class Classification** problem, predict the age of Abalone (a type of Shellfish) using various features provided. Target output is value of Rings (Age = Rings + 1.5).<br />
The dataset can be found [**here**](https://archive.ics.uci.edu/ml/datasets/abalone)<br />
The given features are:
1. Sex
2. Length
3. Diameter
4. Height
5. Whole weight
6. Shucked weight
7. Viscera weight
8. Shell weight

### Details
- Technical Specifications: python, pandas, numpy, scikit-learn
- Type of issue: multiple issues
- Time Limit: **2 days** for each classifier implementation

### Issue requirements / progress
Train the model on training data. Predict target values using test data. Use **Accuracy** and **F1_score** metrics to evaluate the performance.
**Note**: Each pull request should be a solution using only one model.

- [x] Using Logistic Regression
- [x] Using Decision Tree Classifier
- [ ] Using Gaussian Naive Bayes
- [x] Using K-Nearest Neighbor Classifier
- [ ] Using Support Vector Machine Classifier
- [x] Using Random Forest Classifier
- [x] Using Multi-layer Perceptron (MLP) Classifier
- [ ] Using Quadratic Discriminant Analysis 
- [ ] Using AdaBoost Classifier

### Resources
- To help you get started, the starter code ```abalone_age_starter_code.ipynb``` is present in ```/machineLearning/abalone_age```
- Jupyter notebook [installation](https://jupyter.org/install.html)
- Python [tutorial](https://docs.python.org/3/tutorial/index.html)
- Numpy [tutorial](https://www.tutorialspoint.com/numpy/index.htm)
- Pandas [tutorial](https://www.tutorialspoint.com/python_pandas/)
- Scikit-learn [Logistic Regression](https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.LogisticRegression.html)
- Scikit-learn [Decision Tree Classifier](https://scikit-learn.org/stable/modules/generated/sklearn.tree.DecisionTreeClassifier.html)
- Scikit-learn [Gaussian Naive Bayes](https://scikit-learn.org/stable/modules/generated/sklearn.naive_bayes.GaussianNB.html)
- Scikit-learn [K-Nearest Neighbours](https://scikit-learn.org/stable/modules/generated/sklearn.neighbors.KNeighborsClassifier.html)
- Scikit-learn [Support Vector Machines](https://scikit-learn.org/stable/modules/generated/sklearn.svm.SVC.html)
- Scikit-learn [Random Forest Classifier](https://scikit-learn.org/stable/modules/generated/sklearn.ensemble.RandomForestClassifier.html)
- Scikit-learn [MLP Classifier](https://scikit-learn.org/stable/modules/generated/sklearn.neural_network.MLPClassifier.html)
- Scikit-learn [Quadratic Discriminant Analysis](https://scikit-learn.org/stable/modules/generated/sklearn.discriminant_analysis.QuadraticDiscriminantAnalysis.html)
- Scikit-learn [AdaBoost Classifier](https://scikit-learn.org/stable/modules/generated/sklearn.ensemble.AdaBoostClassifier.html)

### Directory Structure
Place your solution file in **path** as follows.
- For Logistic Regression:
 ```/machineLearning/abalone_age/lr/<your_solution_file>```
- For Decision Tree Classifier:
 ```/machineLearning/abalone_age/dtc/<your_solution_file>```
- For Gaussian Naive Bayes:
 ```/machineLearning/abalone_age/gnb/<your_solution_file>```
- For K Neighbors Classifier:
 ```/machineLearning/abalone_age/knn/<your_solution_file>```
- For Support Vector Classifier:
 ```/machineLearning/abalone_age/svc/<your_solution_file>```
- For Using Random Forest Classifier:
```/machineLearning/abalone_age/rfc/<your_solution_file>```
- For Using Multi-layer Perceptron (MLP) Classifier:
```/machineLearning/abalone_age/mlp/<your_solution_file>```
- For Using Quadratic Discriminant Analysis:
```/machineLearning/abalone_age/qda/<your_solution_file>```
- For Using AdaBoost Classifier:
```/machineLearning/abalone_age/adb/<your_solution_file>```

#### Note
Please **claim** the issue first by commenting here before starting to work on it.
For any further issues, feel free to contact me @vrn25 or @niranjanrao1509 any time. Requests for review are welcomed. **Happy Learning!**

**Comment by kampaitees** [comment]
I'm doing it

**Comment by vrn25** [comment]
@kampaitees each model requires a new pull request. Please choose and specify one of the models you would like to work on.

**Comment by kampaitees** [comment]
I had used decision tree classifier

**Comment by Shuddhatm22** [comment]
i want to take up this issue using K neighbours clasiification

**Comment by vrn25** [comment]
@Shuddhatm22 yes please start working on it.

**Comment by i-m-vivek** [comment]
I want to take up this issue using Multi-layer Perceptron (MLP) Classifier.

**Comment by vrn25** [comment]
@i-m-vivek Go ahead

**Comment by i-m-vivek** [comment]
Added MLP Classifier #232 please review it.

**Comment by shashikanthgk** [comment]
I am taking up random forest classifier

**Comment by vrn25** [comment]
@shashikanthgk, Go ahead

**Comment by ArquiteturaJoel** [comment]
I would like to take up this problem -  Logistic Regression

**Comment by vrn25** [comment]
@ArquiteturaJoel, Go ahead

**Comment by SaurabhAgarwala** [comment]
Hacktoberfest 2019 ends. Closing the issue.
